meta/autoid/autoid.go

```diff
@@ -123,9 +164,9 @@ func (alloc *allocator) Rebase(tableID, requiredBase int64, allocIDs bool) error
 		}
 		if allocIDs {
 			newBase = mathutil.MaxInt64(currentEnd, requiredBase)
-			newEnd = newBase + step
+			newEnd = mathutil.MinInt64(math.MaxInt64-step, newBase) + step
 		} else {
-			if currentEnd >= requiredBase {
+			if !alloc.isUnsigned && currentEnd >= requiredBase {
```

Comment:
ditto.

Reply:
!alloc.isUnsigned will always be true here?